Sources/Sentry/SentryOutOfMemoryScopeObserver.m

```diff
@@ -75,15 +75,19 @@ - (void)switchFileHandle
 
 - (void)store:(NSData *)data
 {
-    [self.fileHandle seekToEndOfFile];
-    [self.fileHandle writeData:data];
-    [self.fileHandle writeData:[@"\n" dataUsingEncoding:NSASCIIStringEncoding]];
-
-    self.breadcrumbCounter += 1;
-
-    if (self.breadcrumbCounter >= self.maxBreadcrumbs) {
-        [self switchFileHandle];
-        self.breadcrumbCounter = 0;
+    @try {
+        [self.fileHandle seekToEndOfFile];
+        [self.fileHandle writeData:data];
+        [self.fileHandle writeData:[@"\n" dataUsingEncoding:NSASCIIStringEncoding]];
+
+        self.breadcrumbCounter += 1;
+
+        if (self.breadcrumbCounter >= self.maxBreadcrumbs) {
+            [self switchFileHandle];
+            self.breadcrumbCounter = 0;
```

Comment:
This seems like it should go into a `@finally` block, unless we don’t want to reset the count of an exception occurs? Even switching file handles might help the next try work.

Reply:
Yep good catch 💯